codegen/src/main/java/software/amazon/awssdk/codegen/poet/auth/scheme/EndpointBasedAuthSchemeProviderSpec.java

```diff
@@ -62,18 +63,25 @@ public ClassName className() {
 
     @Override
     public TypeSpec poetSpec() {
-        return PoetUtils.createClassBuilder(className())
-                        .addModifiers(Modifier.PUBLIC, Modifier.FINAL)
-                        .addAnnotation(SdkInternalApi.class)
-                        .addSuperinterface(authSchemeSpecUtils.providerInterfaceName())
-                        .addMethod(constructor())
-                        .addField(defaultInstance())
-                        .addField(modeledResolverInstance())
-                        .addField(endpointDelegateInstance())
-                        .addMethod(createMethod())
-                        .addMethod(resolveAuthSchemeMethod())
-                        .addMethod(endpointProvider())
-                        .build();
+        Map<List<String>, AuthSchemeCodegenMetadata> operationsToSigv4 =
```

Comment:
I'm surprised there are no codegen tests to cover this class or these changes. Can we add?

Reply:
Discussed offline - we can't test in codegen because the auth type / service is hardcoded as "s3"